source/guides/actions/control-flow.md

```diff
@@ -171,6 +171,62 @@ module Web::Controllers::Dashboard
 end
 ```
 
+When `#halt` is used, **Hanami** renders a default status page with the HTTP status and the message.
+
+<p><img src="/images/default-template.png" alt="Hanami default template" class="img-responsive"></p>
+
+To customize the UI for the HTTP 404 error, you can use a [custom error page](/guides/views/custom-error-pages).
+
+## HTTP Status
+
+In case you want let the view to handle the error, instead of using `#halt`, you should use `#status=`.
+
+The typical case is a **failed form submission**: we want to return a non-successful HTTP status (`422`) and let the view to render the form again and show the validation errors.
+
+```ruby
+# apps/web/controllers/books/create.rb
+module Web::Controllers::Books
+  class Create
+    include Web::Action
```

Comment:
@vyper If you're referring to https://github.com/hanami/controller/pull/160, this won't be included yet. Before to merge that PR, we need to make that change in all the libs, and it's a huge effort.